### PR TITLE
finding git root up in the directory  tree

### PR DESCRIPTION
**Pull request by halex2005**
Hello!
I'd like to use Cake.Git tool in my build scripts, but I found that all repositoryDirectoryPath parameters must point to root git working directory.
My cake build script placed inside src subdirectory and I need to find git root up the directories tree.
This PR does just that.
Please, consider to approve this PR.

/cc: @devlead

### CODE REVIEW

**Comment by devlead** [comment]
Hi @halex2005, 

Sorry we've been very busy with other projects, we'll review this PR as soon as possible.
If possible please add a test case too [test.cake](https://github.com/WCOMAB/Cake_Git/blob/master/test.cake)

File: src/Cake.Git/GitAliases.GitFindRootFromPath.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        /// <summary>
+        /// Finding git root path from subtree.
+        /// </summary>
+        /// <param name="context">The context.</param>
+        /// <param name="path">Path to probe the repository.</param>
+        /// <returns>The path to the repository root.</returns>
+        /// <exception cref="ArgumentNullException">If any of the parameters are null.</exception>
+        [CakeMethodAlias]
+        [CakeAliasCategory("FindRootFromPath")]
+        public static string GitFindRootFromPath(this ICakeContext context, string path)
```
Comment (devlead) [inline src/Cake.Git/GitAliases.GitFindRootFromPath.cs]: * Path shouldn't be string but DirectoryPath

File: src/Cake.Git/GitAliases.GitFindRootFromPath.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        {
+            var fsPath = path;
```
Comment (devlead) [inline src/Cake.Git/GitAliases.GitFindRootFromPath.cs]: * Parameters should be null checked [example](https://github.com/WCOMAB/Cake_Git/blob/master/src/Cake.Git/GitAliases.Add.cs#L29)
* Validate path existence with `context.FileSystem.Exist` method.
* Avoid using System.IO in general, instead use Cake Core IO classed, example [here](https://github.com/WCOMAB/Cake_Git/blob/master/src/Cake.Git/GitAliases.Clone.cs#L49)

**Comment by halex2005** [comment]
I rebased this PR to lastest upstream.
Please, review and consider to accept this PR.
Actually, this alias is not necessary for me now. This is just convinient utility function that can be helpful for someone else.
So, descision to include this PR is up to you.
Thanks.

File: src/Cake.Git/GitAliases.GitFindRootFromPath.cs (hunk 2 above)
Comment (halex2005) [inline src/Cake.Git/GitAliases.GitFindRootFromPath.cs]: done

File: src/Cake.Git/GitAliases.GitFindRootFromPath.cs (hunk 1 above)
Comment (halex2005) [inline src/Cake.Git/GitAliases.GitFindRootFromPath.cs]: yep, done

**Comment by devlead** [comment]
@halex2005 LGTM :+1: Merged! thanks for your contribution.

**Pull request merged**